experimental/segmenter/src/line_breaker.rs

```diff
@@ -0,0 +1,849 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+extern crate unicode_width;
+
+use crate::lb_define::*;
+use crate::lstm::*;
+use crate::property_table::*;
+use crate::rule_table::*;
+
+use core::char;
+use core::str::CharIndices;
+use unicode_width::UnicodeWidthChar;
+
+#[derive(Copy, Clone, PartialEq)]
+pub enum LineBreakRule {
+    /// Use `line-break: normal;` line break rule
+    Normal,
+    /// Use `line-break: strict;` line break rule
+    Strict,
+    /// Use `line-break: loose;` line break rule
+    Loose,
+    /// Use `line-break: anywhere;` line break rule
+    Anywhere,
+}
+
+#[derive(Copy, Clone, PartialEq)]
+pub enum WordBreakRule {
+    /// Use `word-break: normal;` line break rule
+    Normal,
+    /// Use `word-break: break-all;` line break rule
+    BreakAll,
+    /// Use `word-break: keep-all;` line break rule
+    KeepAll,
+}
+
+fn get_linebreak_property_utf32_with_rule(
+    codepoint: u32,
+    line_break_rule: LineBreakRule,
+    word_break_rule: WordBreakRule,
+    _ja_zh: bool,
+) -> u8 {
+    let codepoint = codepoint as usize;
```

Comment:
suggestion: All of the conditions here seem to be running roughly the same code, seems like this can be simplified to just `let prop = get_linebreak_property_latin1(..);` followed by a single `if` condition that maps `CJ` to `ID`